packages/fxa-settings/src/components/ThirdPartyAuth/index.tsx

```diff
@@ -147,48 +81,143 @@ const ThirdPartyAuth = ({ enabled = false }: ThirdPartyAuthProps) => {
-                redirectUri: config.appleAuthConfig.redirectUri,
-                responseType: 'code id_token',
-                responseMode: 'form_post',
-              });
+          />
+          <ThirdPartySignInForm
+            {...{
+              party: 'apple',
+              ...config.appleAuthConfig,
+              state: '',
+              scope: 'email',
+              responseType: 'code id_token',
+              accessType: 'offline',
+              prompt: 'consent',
+              responseMode: 'form_post',
+              onSubmit: onContinueWithApple,
+              buttonText: (
+                <>
+                  <AppleLogo />
+                  <FtlMsg id="continue-with-apple-button">
+                    Continue with Apple
+                  </FtlMsg>
+                </>
+              ),
             }}
-          >
-            <AppleLogo />
-            <FtlMsg id="continue-with-apple-button">Continue with Apple</FtlMsg>
-          </button>
+          />
         </div>
       </div>
     </>
   );
 };
 
+/**
+ * Represents the sign in form posted to google third party auth.
+ * Note that the innerRef is used by the parent component to trigger
+ * a form submission.
+ */
+const ThirdPartySignInForm = ({
+  party,
+  authorizationEndpoint,
+  state,
+  clientId,
+  scope,
+  redirectUri,
+  accessType,
+  prompt,
+  responseType,
+  responseMode,
+  buttonText,
+  onSubmit,
+}: {
+  party: 'google' | 'apple';
+  authorizationEndpoint: string;
+  state: string;
+  clientId: string;
+  scope: string;
+  redirectUri: string;
+  accessType: string;
+  prompt: string;
+  responseType: string;
+  responseMode?: string;
+  buttonText: ReactElement;
+  onSubmit?: FormEventHandler<HTMLFormElement>;
+}) => {
+  const stateRef = useRef<HTMLInputElement>(null);
+
+  function onClick() {
+    clearStoredParams();
+    stateRef.current!.value = getState();
+  }
+
+  if (onSubmit === undefined) {
+    onSubmit = () => true;
+  }
+
+  return (
+    <form action={authorizationEndpoint} method="GET" onSubmit={onSubmit}>
+      <input
+        data-testid={`${party}-signin-form-state`}
+        ref={stateRef}
+        type="hidden"
+        name="state"
+        value={state}
+      />
+      <input type="hidden" name="client_id" value={clientId} />
+      <input type="hidden" name="scope" value={scope} />
+      <input type="hidden" name="redirect_uri" value={redirectUri} />
+      <input type="hidden" name="access_type" value={accessType} />
+      <input type="hidden" name="prompt" value={prompt} />
+      <input type="hidden" name="response_type" value={responseType} />
+      {responseMode && (
+        <input type="hidden" name="response_mode" value={responseMode} />
+      )}
+
+      <button
+        type="submit"
+        className="w-full mt-2 justify-center text-black bg-transparent border-black border hover:border-grey-300 font-medium rounded-lg text-sm text-center inline-flex items-center"
```

Comment:
This is more of a nit since it's existing styles from before your PR and I'm not sure how much we worked with Product/UX on these in content-server but there some minor stylistic differences between content-server (first screenshot) and settings:

<img src="https://github.com/mozilla/fxa/assets/13018240/4c1e84ab-9193-437f-81b2-81f60f861708" width="300">
<img width="300" alt="image" src="https://github.com/mozilla/fxa/assets/13018240/f232bdd1-00e3-4a5e-9118-38b892ab66f6">

Reply:
Ah yes, that is definitely a difference. Do you think I should file follow up for it? 

_This sprint is a bit tight, so I'd rather leave it as is since it was already in place, and I'd have to go figure out what the corresponding tail wind would be._

Reply:
I vote for a follow up, or at least adding the slightly larger font for readability :)

A related question that came up while I was looking at the signin flow - are there situations where we might only show ThirdPartyAuth and no password (e.g., for password-less account), or is the ThirdPartyAuth always offered _with_ a password input (whether or not the account has a password)? 

Essentially wondering if the "OR" divider should be optional. (Not requesting a change in this PR, but wondering about implementation in the signin flow)

Reply:
Follow up seems good to me!

> Essentially wondering if the "OR" divider should be optional.

[It is now](https://github.com/mozilla/fxa/pull/15667#discussion_r1291665893) 😎

Reply:
Oh snap, I'd missed that it was added! Sweet. Thanks @dschom!